scripts/autotools/CheckSSL.m4

```diff
@@ -178,6 +178,7 @@ AC_MSG_RESULT([$PHP_MONGODB_SSL])
 dnl Disable Windows SSL and crypto
 AC_SUBST(MONGOC_ENABLE_SSL_SECURE_CHANNEL, 0)
 AC_SUBST(MONGOC_ENABLE_CRYPTO_CNG, 0)
+AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_CNG, 0)
```

Comment:
Rather than modify libmongoc's `CheckSSL.m4` file, I wonder if we can split the libmongocrypt bits out into their own file. I was playing around with the following, which I included _after_ libmongoc's `CheckSSL.m4`:

```m4
if test "$PHP_CLIENT_SIDE_ENCRYPTION" != "no"; then
  AC_MSG_CHECKING([which crypto library to use for libmongocrypt])
  AC_MSG_RESULT([$PHP_MONGODB_SSL])

  dnl Disable Windows crypto
  AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_CNG, 0)

  if test "$PHP_MONGODB_SSL" = "darwin"; then
    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO, 1)
    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_OPENSSL, 0)
    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_COMMON_CRYPTO, 1)
  elif test "$PHP_MONGODB_SSL" = "openssl" -o "$PHP_MONGODB_SSL" = "libressl"; then
    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO, 1)
    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_OPENSSL, 1)
    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_COMMON_CRYPTO, 0)
  fi
fi
```

With this, we would then move the `PHP_MONGODB_ADD_SOURCES` calls back to `config.m4` (with their own `$PHP_MONGODB_SSL` conditions to handle Darwin and OpenSSL/LibreSSL. That would avoid blending libmongoc and libmongocrypt build scripts, and also ensure some consistency since no other scripts under `scripts/autotools` add sources directly (IIRC).

@kevinAlbs: As a side note, I originally worried that libmongocrypt didn't support LibreSSL until I realized both LibreSSL and OpenSSL share libcrypto. I'd propose the "openssl" terminology in libmongocrypt constants and source files get renamed to refer to libcrypto is that wouldn't be backwards breaking (otherwise, no worries). For instance, `CRYPTO_LIBCRYPTO` over `CRYPTO_OPENSSL`.

Reply:
Rename is in the CR for MONGOCRYPT-218.

Reply:
Moving to separate file sounds good 👍

Reply:
Moved a whole bunch of libmongocrypt code to separate files.